scripts/analyze_coupling.py

```
@@ -0,0 +1,142 @@
+        timeseries[var_name] = np.array([np.mean(step_map[s]) for s in steps])
+
+    return timeseries
+
+
+def cross_correlation(x: np.ndarray, y: np.ndarray, max_lag: int) -> list[dict]:
+    """Compute time-lagged Pearson and Spearman correlations.
+
+    Positive lag means x leads y (x[t] correlates with y[t+lag]).
+    """
+    results = []
+    n = len(x)
```

Ruff (0.15.0)

[error] Local variable `n` is assigned to but never used

Remove assignment to unused variable `n`

(F841)